HousingResidentInformationAPI.Tests/V1/UseCase/ValidatePostcodeTests.cs

```diff
@@ -22,6 +22,11 @@ public void ValidPostcodesReturnTrue(string postcode)
         }
 
         [TestCase("1")]
+        [TestCase("111111")]
+        [TestCase("111 111")]
+        [TestCase("EEEEEE")]
+        [TestCase("EEE EEE")]
```

Comment:
If you were to look closely at your [Regex](https://github.com/LBHackney-IT/housing-resident-information-api/blob/d9ea99c598620947e99a493a387470612414349a/HousingResidentInformationAPI/V1/UseCase/ValidatePostcode.cs#L11-L12):
`^([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9]?[A-Za-z])))) [0-9][A-Za-z]{2})$`

you will notice that there's a bug, that allows partial matches due to misplaced "^" start of the line symbol. This first part of the Regex `^([Gg][Ii][Rr] 0[Aa]{2})|` makes it so the start of the line for the pattern in only considered for the "GIR 0AA" match, and for the rest of the postcode patterns that aren't this 1 special case. In order to fix you bug you need to put ^ symbol before your 2nd capture group like so:
`^([Gg][Ii][Rr] 0[Aa]{2})|^((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9]?[A-Za-z])))) [0-9][A-Za-z]{2})$`